connectors/rocketmq-replicator/src/main/java/org/apache/rocketmq/replicator/RmqSourceReplicator.java

```diff
@@ -188,7 +195,7 @@ public Class<? extends Task> taskClass() {
     }
 
     @Override
-    public List<KeyValue> taskConfigs() {
+    public List<KeyValue> taskConfigs(int maxTasks) {
         if (!configValid) {
```

Comment:
maxTasks can specify the number of tasks. Can you consider whether it needs to be divided into multiple tasks?

Reply:
For replicators, taskParallelism is used to control the number of tasks. The taskParallelism field and maxTask are semantically identical, so there is no definition for maxTask.

Reply:
> maxTasks can specify the number of tasks. Can you consider whether it needs to be divided into multiple tasks?

For replicators, taskParallelism is used to control the number of tasks. The taskParallelism field and maxTask are semantically identical, so there is no definition for maxTask.

Reply:
The limitation of the previous version of the connect api, if the replicator wants to provide the ability of multiple tasks, only the replicator itself can provide the number of tasks, but we found that the ability of multiple tasks is general, and the replicator should not implement its own definition configuration, almost every specific connector implementation There is such a demand, so the specified configuration of the number of tasks should be unified. It is easier to use the maxTasks provided by the api to unify this capability. The connector shards the tasks according to the specific storage sharding. The actual number of task shards should be, For example, by topic, min (the number of tasks divided by topic, maxTasks) is configured.

Reply:
> The limitation of the previous version of the connect api, if the replicator wants to provide the ability of multiple tasks, only the replicator itself can provide the number of tasks, but we found that the ability of multiple tasks is general, and the replicator should not implement its own definition configuration, almost every specific connector implementation There is such a demand, so the specified configuration of the number of tasks should be unified. It is easier to use the maxTasks provided by the api to unify this capability. The connector shards the tasks according to the specific storage sharding. The actual number of task shards should be, For example, by topic, min (the number of tasks divided by topic, maxTasks) is configured.

Then I think the taskParallelism parameter should be removed and use maxTask instead of taskParallelism to control the number of tasks. Otherwise, writing like min (the number of tasks divided by topic, maxTasks) will only increase the cost of understanding, what do you think?

Reply:
> 

1、Yes，maxTask should replace taskParallelism.
2、min (number of shards, maxTasks) should also be implemented. This is also easier to understand. If there are only 2 shards and maxTasks=5, the number of tasks can only be 2, which is why the number of tasks is maxTasks instead of taskNum. For example, the data source has only 2 topics, or 2 tables, maxTasks=5, the dimension of the shard is topic or table, and the final task data is 2, min(topicNum, maxTasks)

Reply:
> 

I understand what you mean, but the current replicator uses taskParallelism to do sharding. According to what you said, if 2 topics and taskParallelism is 1, then both topics are on 1 task (not 2 tasks). In this way, taskParallelism may be retained, then the logic is taskParallelism = min(taskParallelism, maxTasks)

Reply:
There is no problem with multiple topics in one task. The main problem is that the configuration of the number of tasks can be unified. Basically, all connectors need this configuration.

Reply:
> There is no problem with multiple topics in one task. The main problem is that the configuration of the number of tasks can be unified. Basically, all connectors need this configuration.

I tried to submit a version, can you review it?